thoth/common/openshift.py

```diff
@@ -1480,18 +1496,51 @@ def schedule_adviser(
                 "Unable to schedule adviser without backend namespace being set"
             )
 
-        job_id = job_id or self.generate_id("adviser")
-        parameters = locals()
-        parameters.pop("self", None)
-        return self._schedule_workload(
-            run_method_name=self.run_adviser.__name__,
-            run_method_parameters=parameters,
-            template_method_name=self.get_adviser_template.__name__,
-            job_id=job_id,
-            namespace=self.backend_namespace,
-            labels={"component": "adviser"},
+        if not self.use_argo:
+            job_id = job_id or self.generate_id("adviser")
+            parameters = locals()
+            parameters.pop("self", None)
+            return self._schedule_workload(
+                run_method_name=self.run_adviser.__name__,
+                run_method_parameters=parameters,
+                template_method_name=self.get_adviser_template.__name__,
+                job_id=job_id,
+                namespace=self.backend_namespace,
+                labels={"component": "adviser"},
+            )
+
+        import workflows
+        adviser_id = job_id or self.generate_id("adviser")
+        template_parameters = {}
+        template_parameters["THOTH_ADVISER_JOB_ID"] = adviser_id
+        template_parameters["THOTH_ADVISER_REQUIREMENTS"] = application_stack["requirements"]
+        template_parameters["THOTH_ADVISER_REQUIREMENTS_LOCKED"] = application_stack["requirements_lock"]
+        template_parameters["THOTH_ADVISER_LIBRARY_USAGE"] = json.dumps(library_usage)
+        template_parameters["THOTH_ADVISER_REQUIREMENTS_FORMAT"] = "pipenv"
+        template_parameters["THOTH_ADVISER_RECOMMENDATION_TYPE"] = recommendation_type
+        template_parameters["THOTH_ADVISER_RUNTIME_ENVIRONMENT"] = runtime_environment
+        template_parameters["THOTH_ADVISER_LIMIT"] = limit
+        template_parameters["THOTH_ADVISER_COUNT"] = count
+        template_parameters["THOTH_ADVISER_LIMIT_LATEST_VERSIONS"] = limit_latest_versions
+
+        workflow_parameters = {}
+        workflow_parameters["THOTH_DOCUMENT_ID"] = f"advise-{adviser_id}"
```

Comment:
Just:

```python
        workflow_parameters["THOTH_DOCUMENT_ID"] = adviser_id
```

Or will it be something like `advise-adviser-401a385e`?

Reply:
From the workflow in adviser workflow template, we have objects:
  - apiVersion: argoproj.io/v1alpha1
    kind: Workflow
    metadata:
      name: "advise-${THOTH_ADVISER_JOB_ID}"
      annotations:
        thoth-station.ninja/template-version: 0.5.0
      labels:
        app: thoth
        component: adviser
    spec:
      serviceAccountName: argo
      podGC:
        strategy: OnWorkflowSuccess
      volumes:
        - name: cache
          emptyDir: {}
      entrypoint: advise
      arguments:
        parameters:
          - name: THOTH_ADVISER_JOB_ID
            value: "${THOTH_ADVISER_JOB_ID}"
          - name: THOTH_DOCUMENT_ID
            value: "advise-${THOTH_ADVISER_JOB_ID}"

Reply:
To integrate this correctly with graph-sync, we will probably need to discard `advise-` prefix - why is it needed?

The point here I'm making - graph-sync-job gets prefix of the document name to check what syncing handler should be run:

https://github.com/thoth-station/storages/blob/master/thoth/storages/sync.py#L457-L465

If the resulting document will be named `advise-adviser-401a385e` (that is IMHO not nice name), graph-sync-job fails to find the correct handler.

Reply:
I agree, so I will adjust the adviser workflow template as well :) Thanks!!